llvm/include/llvm/Transforms/Utils/LoopUtils.h

```diff
@@ -283,6 +284,22 @@ MDNode *updateIterCounts(LLVMContext &Context, MDNode *LoopID,
                          std::function<int64_t(int64_t)> FixMin,
                          std::function<int64_t(int64_t)> FixMax);
 
+/// Get Minimum Trip Count from the LoopID
+std::optional<int64_t> getMinTripCount(const MDNode *LoopID);
+
+/// Get the LoopID from a single block loop or nullptr
+const MDNode *getLoopID(const MachineBasicBlock &LoopBlock);
+
+/// Get LoopID for Latch
+const MDNode *getLoopID(const BasicBlock *LoopBB);
+
+/// LoopBlock should hold the backedge. Since we generally call it for
+/// single block loops, that's automatically true.
+std::optional<int64_t> getMinTripCount(const MachineBasicBlock &LoopBlock);
+
```

Comment:
I think we can handle everything here using `class BasicBlock`. In the backend we can have a wrapper function that uses *[getBasicBlock](https://llvm.org/doxygen/classllvm_1_1MachineBasicBlock.html#a4874816314c3308be0bf1e71de2078d8)(). I agree that this is the place for IR-only code.

Reply:
removed the wrapper